java/java.source.base/src/org/netbeans/api/java/source/SourceUtils.java

```diff
@@ -1127,9 +1134,8 @@ private static boolean isCaseSensitive () {
     }
 
     /**
-     * Returns candidate filenames given a classname. The return value is either
-     * a String (top-level class, no $) or List&lt;String> as the JLS permits $ in
-     * class names.
+     * Returns candidate filenames given a classname.
+     * @return a single name (top-level class, no $) or multiple as the JLS permits $ in class names.
```

Comment:
#5152 made this method type-safe but neglected to adjust the Javadoc accordingly.